src/ChildProcess.php

```
@@ -147,7 +149,9 @@ protected function fromRuntimeProcess($process)
         }
 
         foreach ($process['settings'] as $key => $value) {
-            $this->{$key} = $value;
+            if (property_exists($this, $key)) {
```

I like this, but do we want to bail early and throw an exception if something unexpected is provided? Silently dropping something without a log entry at minimum could make things hard to troubleshoot in future if say a property was accidentally removed in subsequent PRs

---

Updated the PR. Tell me what you think.